### ISSUE TITLE
problem registering same taxonomy for different post type

### ISSUE THREAD

**Issue by webgrafia** [jjgrainger/PostTypes#6]
Hi, I love your tool, but in this version there's a problem when registering the same custom taxonomy for different post type.
example:

```
  $gallery_options = [
        'has_archive' => true,
        'supports' => array("title", "editor", "author", "thumbnail"),
        'menu_position' => 5
    ];
    $gallery_names = [
        'name' => 'gallery',
        'singular' => 'Gallery',
        'plural' => 'Gallery',
        'slug' => 'gallery'
    ];
    $gallery = new PostType($gallery_names, $gallery_options);

 $marca_names = [
        'name' => 'marca',
        'singular' => 'Marca',
        'plural' => 'Marche',
        'slug' => 'marca'
    ];
    $marca_options = [
        'hierarchical' => false
    ];
$gallery->taxonomy($marca_names, $marca_options);
...
 $video = new PostType($video_names, $video_options);
 $video->taxonomy("marca");

```

I think the problem is in
`if (taxonomy_exists($names['name'])) {
            $this->existingTaxonomies[] = $names['name'];`

in taxonomy function, it return false.

Do you experienced the same issue?

**Comment by jjgrainger** [comment]
Thanks for pointing this out, I'll get to fixing this as soon as I can :)

**Comment by MWDelaney** [comment]
I can confirm this problem --in fact trying to attach an existing taxonomy like `post_tag` actually _removes_ it from posts.

**Comment by cedarstay** [comment]
Can confirm that I am also experiencing this issue

**Comment by jjgrainger** [comment]
Hi @cedarstay @MWDelaney and @webgrafia,

Just thought I'd let you know this has been fixed now. :)

The `taxonomy()` method was trying to use WordPress `taxonomy_exists` function to check if the taxonomy existed. This call was made outside of any actions, so was being run before WordPress was fully loaded. This meant the `taxonomy_exists` check didn't work properly.

I've updated the code so the taxonomy names and options are saved. Then the checks and registration of the taxonomies are run at the correct points in the WordPress event flow.

Thanks everyone 👍
